### PR TITLE
[W6.2i][T16-1]Gian Zhi Cai

### PR DESCRIPTION
**Pull request by zhicaizack**
Use interfaces [LO-Interfaces]
Used interface method getPrintableString for the command 'view'

### CODE REVIEW

File: src/seedu/addressbook/commands/ViewCommand.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public CommandResult execute() {
             if (!addressBook.containsPerson(target)) {
                 return new CommandResult(Messages.MESSAGE_PERSON_NOT_IN_ADDRESSBOOK);
             }
-            return new CommandResult(String.format(MESSAGE_VIEW_PERSON_DETAILS, target.getAsTextHidePrivate()));
+            return new CommandResult(String.format(MESSAGE_VIEW_PERSON_DETAILS, target.getPrintableString(target.getName(), target.getPhone(), target.getEmail(), target.getAddress())));
```
Comment (stephlewyh) [inline src/seedu/addressbook/commands/ViewCommand.java:35]: You used the newly implemented `getPrintableString(...)` - great!

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -55,4 +56,9 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+    @Override
+    public String getPrintableString() {
+        return isPrivate() ? "" : String.format(MESSAGE_PERSON_ADDRESS, value);
```
Comment (stephlewyh) [inline src/seedu/addressbook/data/person/Address.java:62]: good use of ternary expression here.

Comment (stephlewyh) [reply src/seedu/addressbook/data/person/Address.java:62]: Nice try with this. You may also consider formatting strings with `Formatter`, from java.util.Formatter.

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -55,4 +56,9 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+    @Override
```
Comment (stephlewyh) [inline src/seedu/addressbook/data/person/Address.java:60]: You may apply a static modifier to `getPrintableString(Printable printables)`. A static method invoked without the need for creating an instance of a class, and the method is stored in one memory address.

**Comment by zhicaizack** [comment]
Noted with thanks 💯

**Pull request closed without merging**